src/lib/y2partitioner/widgets/configure.rb

```diff
@@ -0,0 +1,280 @@
+Yast.import "Popup"
+Yast.import "Arch"
+Yast.import "PackageCallbacks"
+Yast.import "PackageSystem"
+
+module Y2Partitioner
+  module Widgets
+    # "Configure" menu button used to run the corresponding YaST clients to
+    # activate several storage technologies
+    #
+    # Most of the behavior of this widget is a direct translation of the old
+    # Yast::PartitioningEpAllInclude from yast-storage
+    class Configure < CWM::CustomWidget
+      include Reprobe
+
+      # Constructor
+      def initialize
+        textdomain "storage"
+        super
+      end
+
+      # Content of the widget, a menu button with the list of available
+      # configuration clients or an empty widget if no client is available
+      #
+      # @macro seeCustomWidget
+      #
+      # @return [Yast::WidgetTerm]
+      def contents
+        @contents ||= actions.empty? ? Empty() : MenuButton(Opt(*opt), label, items)
+      end
+
+      # Event handler for the configuration menu
+      #
+      # @param event [Hash] UI event
+      # @return [:reprobe, nil] :reprobe when some configuration client was
+      #   executed; nil otherwise.
+      def handle(event)
+        action = Action.find(event["ID"])
+
+        return nil unless action
+        return nil unless warning_accepted?(action) && availability_ensured?(action)
+
+        Yast::WFM.call(action.client)
+        reprobe
+        :reprobe
+      end
+
+    private
+
+      # @return [Array<Yast::WidgetTerm>]
+      def items
+        actions.map do |action|
+          Item(Id(action.id), Yast::Term.new(:icon, action.icon), action.label)
+        end
+      end
+
+      # @return [Array<Symbol>]
+      def opt
+        [:key_F7]
+      end
+
+      # Localized label for the menu button
+      #
+      # @return [String]
+      def label
+        # Translators: Configure menu in the initial Partitioner screen
+        _("Configure...")
+      end
+
+      # @macro seeCustomWidget
+      #
+      # Redefined in this class because the base implementation at CWM::CustomWidget
+      # does not search for ids into the items of a MenuButton.
+      def ids_in_contents
+        actions.map(&:id)
+      end
+
+      # Available actions to be offered to the user
+      #
+      # @return [Array<Action>]
+      def actions
+        @actions ||=
+          if Yast::Stage.initial
+            # In the inst-sys, check which clients are available
+            Action.supported.select { |action| Yast::WFM.ClientExists(action.client) }
+          else
+            # In the installed system, we don't care if the client is there or not
+            # as the user will be prompted to install the pkg anyway (in #handle).
+            Action.supported
+          end
+      end
+
+      # Displays the action warning to the user
+      #
+      # @param action [Action]
+      # @return [Boolean] whether the user confirmed the warning message
+      def warning_accepted?(action)
+        Yast::Popup.YesNo(action.warning_text)
```

Comment:
I guess we should start to always use `Yast2::Popup` in new code.

Reply:
Well, in this particular case it provides no benefit (even the opposite, code is more verbose). Another example of being conservative.